modules/core/src/main/java/org/apache/ignite/internal/processors/rest/client/message/GridClientAuthenticationRequest.java

```diff
@@ -29,34 +32,55 @@
     private static final long serialVersionUID = 0L;
 
     /** Credentials. */
-    private Object cred;
+    private SecurityCredentials cred;
+
+    /** User attributes. */
+    private Map<String, String> userAttrs;
 
     /**
      * @return Credentials object.
      */
-    public Object credentials() {
+    public SecurityCredentials credentials() {
         return cred;
     }
 
     /**
      * @param cred Credentials object.
      */
-    public void credentials(Object cred) {
+    public void credentials(SecurityCredentials cred) {
         this.cred = cred;
     }
 
+    /**
+     * @return User attributes.
+     */
+    public Map<String, String> userAttributes() {
```

Comment:
Please add `@Nullable` annotation to the return type.

Reply:
Done.